.github/workflows/build.yml

```diff
@@ -0,0 +1,115 @@
+---
+name: build
+
+on:
+  push:
+    branches:
+      - trying
+      - staging
+      - master
+  pull_request:
+    branches:
+      - '*'
+jobs:
+  build-test:
+    name: Build and Test
+    runs-on: ubuntu-latest
+    env:
+      RIOT_BRANCH: '2021.01'
+      DOCKER_REGISTRY: "${{ secrets.DOCKERHUB_USERNAME || 'local' }}"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          driver: docker
+
+      - name: Build riotdocker-base
+        uses: docker/build-push-action@v2
+        with:
+          context: ./riotdocker-base
+          tags: |
+            local/riotdocker-base:latest
+            ${{ env.DOCKER_REGISTRY }}/riotdocker-base:latest
+            ${{ env.DOCKER_REGISTRY }}/riotdocker-base:${{ env.RIOT_BRANCH }}
+
+      - name: Build static-test-tools
+        uses: docker/build-push-action@v2
+        with:
+          context: ./static-test-tools
+          tags: |
+            local/static-test-tools:latest
```

Comment:
This line doesn't seem required since you set `DOCKERHUB_REGISTRY=local` in `build_args`.

BTW, why not use the same `build_args` strategy with the `riotdocker-base` image build ?

Reply:
It's not the same, one is used to tag the generated image, the other is used to define the base image for that stage build, so the `FROM` statement in the dockerfile

Reply:
Hmm, ok. So I think I don't understand why the `local/` is needed and why this env is required:

```
DOCKER_REGISTRY: "${{ secrets.DOCKERHUB_USERNAME || 'local' }}"
```

Reply:
When building on a PR, secrets are not passed, so `secrets.DOCKERHUB_USERNAME` does not exist.

Reply:
So, DOCKER_REGISTRY is set to local in this case. So there's no need for this line I guess.

Reply:
Ahhh yes seems like it

Reply:
Done